File: android/src/main/kotlin/com/emarsys/emarsys_sdk/command/predict/TrackSearchTermCommand.kt
package com.emarsys.emarsys_sdk.command.predict

import com.emarsys.Emarsys
import com.emarsys.emarsys_sdk.command.EmarsysCommand
import com.emarsys.emarsys_sdk.command.ResultCallback

class TrackSearchTermCommand : EmarsysCommand {
    override fun execute(parameters: Map<String, Any?>?, resultCallback: ResultCallback) {
        val searchTerm: String? = parameters?.get("searchTerm") as String?
        searchTerm ?: throw IllegalArgumentException("searchTerm should not be null!")

        Emarsys.predict.trackSearchTerm(searchTerm)

        resultCallback.invoke(null, null)
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false
        return true
    }

    override fun hashCode(): Int {
        return javaClass.hashCode()
    }
}

File: ios/Classes/Wrapper/Commands/Predict/TrackSearchTermCommand.swift
//
// Created by Emarsys
//

import EmarsysSDK

class TrackSearchTermCommand: EmarsysCommandProtocol {
    func execute(arguments: [String: Any]?, resultCallback: @escaping ResultCallback) {
        guard let searchTerm = arguments?["searchTerm"] as? String else {
            resultCallback(["error": "Invalid searchTerm argument"])
            return
        }
        
        Emarsys.predict.trackSearch(searchTerm: searchTerm)

        resultCallback(["success": true])
    }
}
